ecosystem/indexer/src/models/write_set_changes.rs

```diff
@@ -0,0 +1,110 @@
+    pub resource: serde_json::Value,
+    pub data: serde_json::Value,
+
+    // Default time columns
+    pub inserted_at: chrono::NaiveDateTime,
+}
+
+impl WriteSetChange {
+    pub fn from_write_set_change(
+        transaction_hash: String,
+        write_set_change: &APIWriteSetChange,
+    ) -> Self {
+        match write_set_change {
+            APIWriteSetChange::DeleteModule {
+                address,
+                state_key_hash,
+                module,
+            } => WriteSetChange {
+                transaction_hash,
+                hash: state_key_hash.to_hex_literal(),
+                type_: write_set_change.type_str().to_string(),
+                address: address.to_string(),
+                module: serde_json::to_value(module).unwrap(),
+                resource: Default::default(),
+                data: Default::default(),
+                inserted_at: chrono::Utc::now().naive_utc(),
+            },
+            APIWriteSetChange::DeleteResource {
+                address,
+                state_key_hash,
+                resource,
+            } => WriteSetChange {
+                transaction_hash,
+                hash: state_key_hash.to_hex_literal(),
+                type_: write_set_change.type_str().to_string(),
+                address: address.to_string(),
+                module: Default::default(),
+                resource: serde_json::to_value(resource).unwrap(),
+                data: Default::default(),
+                inserted_at: chrono::Utc::now().naive_utc(),
+            },
+            APIWriteSetChange::WriteModule {
+                address,
+                state_key_hash,
+                data,
+            } => WriteSetChange {
+                transaction_hash,
+                hash: state_key_hash.to_hex_literal(),
+                type_: write_set_change.type_str().to_string(),
+                address: address.to_string(),
+                module: Default::default(),
+                resource: Default::default(),
+                data: serde_json::to_value(data).unwrap(),
+                inserted_at: chrono::Utc::now().naive_utc(),
+            },
+            APIWriteSetChange::WriteResource {
+                address,
+                state_key_hash,
+                data,
+            } => WriteSetChange {
+                transaction_hash,
+                hash: state_key_hash.to_hex_literal(),
+                type_: write_set_change.type_str().to_string(),
+                address: address.to_string(),
+                module: Default::default(),
+                resource: Default::default(),
+                data: serde_json::to_value(data).unwrap(),
+                inserted_at: chrono::Utc::now().naive_utc(),
+            },
+        }
+    }
+
+    pub fn from_write_set_changes(
+        transaction_hash: String,
+        write_set_changes: &[APIWriteSetChange],
+    ) -> Option<Vec<Self>> {
+        if write_set_changes.is_empty() {
+            return None;
```

Comment:
Should you return `None` here or Some(empty_vector)? Semantically we should return an empty vector right?

Reply:
agreed, but this follows the pattern from the other queries where they can all return `None`, and so callers can ignore the result; a rewrite might work though